### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by Jasha10**
CI on the `main` branch seems to be failing.

### CODE REVIEW

**Comment by jieru-hu** [comment]
thanks Jasha. looks like tests are failing due to the new warnings from importlib-resource's latest release.

maybe for `main` branch we can just update the code and update requirements.txt to be `importlib-resources>5.2.3`

i'm assuming this is going to fail for 1.1_branch as well. for which we could do potentially something similar to what you did here https://github.com/omry/omegaconf/pull/808 to avoid updating the dependency for a stable release. 

Not sure what is the best way here. thoughts?

**Comment by Jasha10** [comment]
It seems that importlib_resources has deprecated one API in favor of another one.
```text
DeprecationWarning: is_resource is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
```

I tried following the advice from the url in the DeprecationWarning, copying
the `importlib_resources._legacy.is_resource` function as a drop-in replacement
for the `resources.is_resource` call that we had before.

Before, the python 3.9 tests were passing and the 3.6/7/8 tests were failing.
Now the 3.9 tests are failing and the 3.6/7/8 tests are passing :)

I'll try to find something that works for all the python versions.

File: hydra/_internal/core_plugins/importlib_resources_config_source.py [hunk 1]
```diff
@@ -63,12 +63,12 @@ def load_config(self, config_path: str) -> ConfigResult:
...
-            ret = resources.is_resource(self.path, "__init__.py")  # type:ignore
+            ret = any(
+                f.name == "__init__.py" and f.is_file() for f in resources.files(self.path).iterdir()  # type: ignore
+            )
             assert isinstance(ret, bool)
             return ret
-        except ValueError:
-            return False
-        except ModuleNotFoundError:
+        except Exception:
```
[the remark this answers is not in the document]
Comment (pixelb) [reply hydra/_internal/core_plugins/importlib_resources_config_source.py:71]: Is it better then to catch `(ValueError, ModuleNotFoundError, TypeError)` ?

[the remark this answers is not in the document]
Comment (Jasha10) [reply hydra/_internal/core_plugins/importlib_resources_config_source.py:71]: Something like that would be fine with me. I was copying the code style that I saw elsewhere in the same python file, e.g. [here](https://github.com/facebookresearch/hydra/blob/045fc425fb26010baaec09c2312183272f3058df/hydra/_internal/core_plugins/importlib_resources_config_source.py#L77) and [here](https://github.com/facebookresearch/hydra/blob/045fc425fb26010baaec09c2312183272f3058df/hydra/_internal/core_plugins/importlib_resources_config_source.py#L89).

[the remark this answers is not in the document]
Comment (Jasha10) [reply hydra/_internal/core_plugins/importlib_resources_config_source.py]: Updated in https://github.com/facebookresearch/hydra/pull/1857/commits/d6eb638cc1f4a5c3406fda16766b46c00f9ec9d1.

**Pull request merged**